### DISCUSSION TITLE
How to use graphql-request's batching via the batchRequests() function in redux-saga?

### DISCUSSION THREAD

**Discussion by rishavbharti** [redux-saga/redux-saga#2285] [Q&A]
I'm trying to implement batching in redux-saga using graphql-request's `batchRequests()` function but not sure of the syntax.

Here's what I've tried

        export default ({ gqlClient } = {}) => {
    	    function* getDashboardData() {
    		while (true) {
    			try {
    				yield take(dashboardActions.getDashboardDataPending.type);
    
                    yield call([gqlClient, gqlClient.batchRequests],
    				   { document: queries.home.getUserData, variables: { period: 'day', date: 'last30' } },
    				   { document: queries.home.getRevenueData, variables: { period: 'day', date: 'last60' } }
    				 );
    				 
    				 ...
    		        }
                }
            }
        }

```
   export default ({ gqlClient } = {}) => {
    	    function* getDashboardData() {
    		while (true) {
    			try {
    				yield take(dashboardActions.getDashboardDataPending.type);
    
                    yield apply(gqlClient, gqlClient.batchRequests,
    				 [
    				   { document: queries.home.getUserData, variables: { period: 'day', date: 'last30' } },
    				   { document: queries.home.getRevenueData, variables: { period: 'day', date: 'last60' } }
    				 ]);
    				 
    				 ...
    		        }
                }
            }
        }
```

But this doesn't work.
[Error screenshot][1]
Using graphql-request v3.7.0 & redux-saga v1.1.3

Can someone tell the correct way to do it?
[graphql-request's docs][2]

  [1]: https://i.stack.imgur.com/09bEQ.png
  [2]: https://github.com/prisma-labs/graphql-request#batching

**Comment by neurosnap** [comment]
The function call you cited doesn't look quite right. Try

```JS
yield call([gqlClient, 'batchRequests'], ...args)
```

This function signature is only required if you need the context of gqlClient. If you don't, then 

```JS
yield call(gqlClient.batchRequests, ...args)
```

Should work as well
